src/lib/core/datetime.h

```diff
@@ -400,6 +401,18 @@ datetime_usec(const struct datetime *date)
 	return datetime_nsec(date) / 1000;
 }
 
+/** Simplified checker for datetime structure validity */
+static inline bool
+datetime_validate(const struct datetime *date)
+{
+	if (unlikely(date->epoch < MIN_EPOCH_SECS_VALUE) ||
+	    unlikely(date->epoch > MAX_EPOCH_SECS_VALUE))
+		return false;
```

Comment:
Good suggestion!

Reply:
```diff
diff --git a/src/lib/core/datetime.h b/src/lib/core/datetime.h
index 725b500ff..63fe45962 100644
--- a/src/lib/core/datetime.h
+++ b/src/lib/core/datetime.h
@@ -62,6 +62,12 @@ struct tnt_tm;
 #define MIN_EPOCH_SECS_VALUE    \
 	(MIN_DT_DAY_VALUE * SECS_PER_DAY - SECS_EPOCH_1970_OFFSET)
 
+/** at the moment the range of known timezones is UTC-12:00..UTC+14:00
+ * https://en.wikipedia.org/wiki/List_of_UTC_time_offsets
+  */
+#define MAX_TZOFFSET (14L*60)
+#define MIN_TZOFFSET (-12L*60)
+
 /**
  * datetime structure keeps number of seconds and
  * nanoseconds since Unix Epoch.
@@ -411,6 +417,9 @@ datetime_validate(const struct datetime *date)
 		return false;
 	if (unlikely(date->nsec < 0) || unlikely(date->nsec > NANOS_PER_SEC))
 		return false;
+	if (unlikely(date->tzoffset < MIN_TZOFFSET) ||
+	    unlikely(date->tzoffset > MAX_TZOFFSET))
+		return false;
 	return true;
 }
 
```